Source/Diagnostics/BTDiagnostics.H

```diff
@@ -179,6 +179,15 @@ private:
      *  will be equal to the predicted m_max_buffer_multifabs for each snapshot.
      */
     amrex::Vector<int> m_max_buffer_multifabs;
+    /** Vector of integers to indicate if the snapshot is full (0 not full, 1 for full).
+     *  If the snapshot is full, then the snapshot files are closed.
+     */
+    amrex::Vector<int> m_snapshot_full;
```

Comment:
Not in this PR but maybe as a clean-up after #1898 is merged: let us change the `int` types to `bool` types where we are using boolean meaning. Using `int` for `bool` is old C-style ([before C99](https://www.geeksforgeeks.org/bool-in-c/), there was no `bool`). Using bool will make it easier to understand the states of I/O for other developers :)

Reply:
I tried that first -- but it wouldn't compile locally for me.
And so changed to integer. it could be because std::vector<bool> has some specialization for space efficiency. 
Let me know if you have work-arounds.
https://stackoverflow.com/questions/61158642/c-i-dont-understand-this-error-error-cannot-bind-non-const-lvalue-refere?noredirect=1&lq=1

Reply:
Do you use `m_snapshot_full` and similar `std::vector<bool>` in MPI comms? Accessing `.data()` like an array won't work for them, but I am not sure we do that.

Reply:
we dont -- but I stored it as amrex::Vector<bool> and it complained. I dont access it for MPI comms. 
There are a number of such vectors that I would prefer to use boolean for, but used integer because I stored them as a vector.

Reply:
Ah, AMReX' vector. Didn't know. Curious :)